src/components/UserInfo.js

```diff
@@ -1,19 +1,32 @@
 class UserInfo {
-  constructor({ nameSelector, jobSelector }) {
+  constructor({ nameSelector, jobSelector, avatarSelector }) {
     this._nameElement = document.querySelector(nameSelector);
     this._jobElement = document.querySelector(jobSelector);
+    this._avatarElement = document.querySelector(avatarSelector);
   }
 
   getUserInfo() {
     return {
       name: this._nameElement.textContent,
       job: this._jobElement.textContent,
+      avatar: this._avatarElement.src,
     };
   }
 
-  setUserInfo({ name, job }) {
-    this._nameElement.textContent = name;
-    this._jobElement.textContent = job;
+  setUserInfo({ name, job, avatar }) {
+    if (name) {
+      this._nameElement.textContent = name;
+    }
+    if (job) {
+      this._jobElement.textContent = job;
+    }
+    if (avatar) {
```

Comment:
Needs correction: this method should not accept the avatar, you should update avatar  separately using setAvatar method